### PR TITLE
HDDS-10208. Simplify tests using assertThrows in hadoop-hdds

### PR DESCRIPTION
**Pull request by wzhallright**
## What changes were proposed in this pull request?
Simplify tests using assertThrows in hadoop-hdds

## What is the link to the Apache JIRA

[HDDS-10208](https://issues.apache.org/jira/browse/HDDS-10208)

## CI
https://github.com/wzhallright/ozone/actions/runs/7737861963

### CODE REVIEW

File: hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestPipelineStateManagerImpl.java [hunk 1]
```diff
@@ -295,37 +295,36 @@ public void testGetPipelinesByTypeFactorAndState()
 
   @Test
   public void testAddAndGetContainer() throws IOException, TimeoutException {
-    long containerID = 0;
```
Comment (fapifta) [inline hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestPipelineStateManagerImpl.java:298]: May I ask what is the rationale behind changing this long to AtomicLong? The rest of the patch is really just changing exception testing to assertThrows, is it by chance a leftover from some other work, or intentional?

**Review by fapifta — COMMENTED** [review]
Thank you for the path @wzhallright the changes look good except that one change I left an inline comment on, can you please elaborate on that?

File: hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestPipelineStateManagerImpl.java (hunk 1 above)
Comment (adoroszlai) [reply hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestPipelineStateManagerImpl.java:298]: I assume it is changed to be able to use in this lambda expression:

https://github.com/apache/ozone/blob/482ec1ac126e9b27b8d4659be56f09785eed28a3/hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestPipelineStateManagerImpl.java#L322-L325

I think it's fine.  If we want to minimize the diff, we could keep `long containerID` and change the code above to something like this:

```java
    ContainerID cid = ContainerID.valueOf(++containerID);
    IOException e = assertThrows(IOException.class,
        () -> stateManager.addContainerToPipeline(finalPipeline.getId(), cid));
```

Comment (wzhallright) [reply hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestPipelineStateManagerImpl.java:298]: @adoroszlai That's a good idea. I will modify it. Thanks

**Review by adoroszlai — APPROVED** [review]

**Comment by adoroszlai** [comment]
Thanks @wzhallright for the patch, @fapifta for the review.

File: hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestPipelineStateManagerImpl.java (hunk 1 above)
Comment (fapifta) [reply hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/pipeline/TestPipelineStateManagerImpl.java]: thank you for the explanation, I did not catch that point. my bad.

**Pull request merged**